auth/user_mgt.go

```diff
@@ -0,0 +1,711 @@
+
+func (p *commonParams) appendErrString(s string, i ...interface{}) {
+	p.errors = append(p.errors, fmt.Sprintf(s, i...))
+}
+
+func (p *commonParams) payloadInitialized() {
+	if p.payload == nil {
+		p.payload = make(map[string]interface{})
+	}
+}
+
+func (p *commonParams) set(key string, value interface{}) {
+	p.payloadInitialized()
+	p.payload[key] = value
+}
+
+// assumes that payloadName is a string field in p.payload
+func processDeletion(p *commonParams, payloadName, errorName string) {
+	if dn, ok := p.payload[payloadName]; ok && len(dn.(string)) == 0 {
+		p.addToListParam(deletionSpecs[payloadName].deleteListName, deletionSpecs[payloadName].deleteFieldName)
+		delete(p.payload, payloadName)
+	}
+}
+
+func processClaims(p *commonParams, payloadName, errorName string) {
+	if _, ok := p.payload[payloadName]; !ok {
+		return
+	}
+	p.checkReservedClaims()
+	p.setCustomAttributes()
+}
+func (p *commonParams) checkReservedClaims() {
+	cc := p.payload["customClaims"]
+	switch claims := cc.(type) {
+	case map[string]interface{}:
+		for _, key := range reservedClaims {
+			if _, ok := claims[key]; ok {
+				p.appendErrString("claim %q is reserved, and must not be set", key)
+			}
+		}
+	}
+}
+
+func (p *commonParams) setCustomAttributes() {
+	cc := p.payload["customClaims"]
+	b, err := json.Marshal(cc)
+	if err != nil {
+		p.appendErrString("invalid custom claims Marshaling error: %v wanted %v", err, cc)
+		return
+	} //else if len(b) > maxLenPayloadCC {
+	//		p.appendErrString("Custom Claims payload must not exceed %d characters", maxLenPayloadCC)
+	//	} else {
+	s := string(b)
+	if s == "null" {
+		s = "{}"
+	}
+}
+
+func (p *commonParams) addToListParam(listname, param string) {
+	if _, ok := p.payload[listname]; ok {
+		p.payload[listname] = append(p.payload[listname].([]string), param)
+	} else {
+		p.set(listname, []string{param})
+	}
+}
+
+// Validators.
+
+// No validation needed. used for bool fields
+func allowed(p *commonParams, _, _ string) {
+	return
+}
+
+func nonEmpty(p *commonParams, fieldName, errorName string) {
+	if val, ok := p.payload[fieldName]; ok {
+		if len(val.(string)) == 0 {
+			p.appendErrString("invalid %s:: %q. The %s must be a non-empty string", errorName, val)
+		}
+	}
+}
+
+func validEmail(p *commonParams, fieldName, errorName string) {
+	if val, ok := p.payload[fieldName]; ok {
+		if parts := strings.Split(val.(string), "@"); len(parts) != 2 || len(parts[0]) == 0 || len(parts[1]) == 0 {
+			p.appendErrString("malformed %s string: %q", val, errorName)
+		}
+	}
+}
+
+func strlenGTE(i int) func(*commonParams, string, string) {
+	return func(p *commonParams, fieldName, errorName string) {
+		if val, ok := p.payload[fieldName]; ok {
+			if len(val.(string)) < i {
+				p.appendErrString("invalid %s string. %s must be a string at least %d characters long", errorName, errorName, i)
+			}
+		}
+	}
+}
+
+func strlenLTE(i int) func(*commonParams, string, string) {
```

Comment:
This type of nesting impacts the readability of the code. Please inline the logic into places where it is needed, and dissolve this function.